library/core/src/slice/mod.rs

```diff
@@ -3440,6 +3432,36 @@ impl<T> [T] {
     }
 }
 
+trait CloneFromSpec<T> {
+    fn spec_clone_from(&mut self, src: &[T]);
+}
+
+impl<T> CloneFromSpec<T> for [T]
+where
+    T: Clone,
+{
+    default fn spec_clone_from(&mut self, src: &[T]) {
+        assert!(self.len() == src.len(), "destination and source slices have different lengths");
+        // NOTE: We need to explicitly slice them to the same length
+        // for bounds checking to be elided, and the optimizer will
+        // generate memcpy for simple cases (for example T = u8).
```

Comment:
This comment could be reworded now.
It's a nice change - I think this method was mentioned as a problem spot - for alternative backends and alternative rust implementations